cli/tools/lint/ast_buffer/ts_estree.rs

```diff
@@ -0,0 +1,513 @@
+      AstProp::Finalizer => "finalizer",
+      AstProp::Flags => "flags",
+      AstProp::Generator => "generator",
+      AstProp::Handler => "handler",
+      AstProp::Id => "id",
+      AstProp::In => "in",
+      AstProp::IndexType => "indexType",
+      AstProp::Init => "init",
+      AstProp::Initializer => "initializer",
+      AstProp::Implements => "implements",
+      AstProp::Key => "key",
+      AstProp::Kind => "kind",
+      AstProp::Label => "label",
+      AstProp::Left => "left",
+      AstProp::Literal => "literal",
+      AstProp::Local => "local",
+      AstProp::Members => "members",
+      AstProp::Meta => "meta",
+      AstProp::Method => "method",
+      AstProp::Name => "name",
+      AstProp::Namespace => "namespace",
+      AstProp::NameType => "nameType",
+      AstProp::Object => "object",
+      AstProp::ObjectType => "objectType",
+      AstProp::OpeningElement => "openingElement",
+      AstProp::OpeningFragment => "openingFragment",
+      AstProp::Operator => "operator",
+      AstProp::Optional => "optional",
+      AstProp::Out => "out",
+      AstProp::Param => "param",
+      AstProp::ParameterName => "parameterName",
+      AstProp::Params => "params",
+      AstProp::Pattern => "pattern",
+      AstProp::Prefix => "prefix",
+      AstProp::Properties => "properties",
+      AstProp::Property => "property",
+      AstProp::Qualifier => "qualifier",
+      AstProp::Quasi => "quasi",
+      AstProp::Quasis => "quasis",
+      AstProp::Raw => "raw",
+      AstProp::Readonly => "readonly",
+      AstProp::ReturnType => "returnType",
+      AstProp::Right => "right",
+      AstProp::SelfClosing => "selfClosing",
+      AstProp::Shorthand => "shorthand",
+      AstProp::Source => "source",
+      AstProp::SourceType => "sourceType",
+      AstProp::Specifiers => "specifiers",
+      AstProp::Static => "static",
+      AstProp::SuperClass => "superClass",
+      AstProp::SuperTypeArguments => "superTypeArguments",
+      AstProp::Tag => "tag",
+      AstProp::Tail => "tail",
+      AstProp::Test => "test",
+      AstProp::TrueType => "trueType",
+      AstProp::TypeAnnotation => "typeAnnotation",
+      AstProp::TypeArguments => "typeArguments",
+      AstProp::TypeName => "typeName",
+      AstProp::TypeParameter => "typeParameter",
+      AstProp::TypeParameters => "typeParameters",
+      AstProp::Types => "types",
+      AstProp::Update => "update",
+      AstProp::Value => "value",
+    };
+
+    write!(f, "{}", s)
+  }
+}
+
+impl From<AstProp> for u8 {
+  fn from(m: AstProp) -> u8 {
+    m as u8
+  }
+}
+
+pub struct TsEsTreeBuilder {
+  ctx: SerializeCtx,
+}
+
+// TODO: Add a builder API to make it easier to convert from different source
+// ast formats.
+impl TsEsTreeBuilder {
+  pub fn new() -> Self {
+    // Max values
+    // TODO: Maybe there is a rust macro to grab the last enum value?
+    let kind_count: u8 = AstNode::TSEnumBody.into();
+    let prop_count: u8 = AstProp::Value.into();
```

Comment:
Is this off by 1 since `Value` is the index I think?

Reply:
Good point, it's currently accounted for here https://github.com/denoland/deno/blob/3ba95cd7edfe90ee0cb248294ae24d904bcb552b/cli/tools/lint/ast_buffer/buffer.rs#L203-L204

Reply:
Yeah, or rename to `prop_max_index` (though probably best to just account for it here)